File: src/main/java/net/dontdrinkandroot/gitki/repository/UserRepository.java
package net.dontdrinkandroot.gitki.repository;

import net.dontdrinkandroot.gitki.model.Role;
import net.dontdrinkandroot.gitki.model.User;
import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.stereotype.Repository;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
@Repository
public interface UserRepository extends PagingAndSortingRepository<User, Long>
{
    Long countByRole(Role role);

    User findByEmail(String username);
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/repository/UserRepository.kt
package net.dontdrinkandroot.gitki.repository

import net.dontdrinkandroot.gitki.model.Role
import net.dontdrinkandroot.gitki.model.User
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : PagingAndSortingRepository<User, Long> {

    fun countByRole(role: Role): Long
    fun findByEmail(username: String): User
}